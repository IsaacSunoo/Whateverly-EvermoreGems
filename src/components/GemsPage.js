import React, { Component } from "react";
import GemCard from "./GemCard";
import Header from "./Header";
import Greeting from "./Greeting";

class GemsPage extends Component {
    constructor() {
        super();
        this.state = {
          gemPage: false,
          searchInput: '',
          searchedGems: []
        };
    }

  updateInputValue = (e) => {
    let searchInput = e.target.value;
    this.setState({
      searchInput: searchInput.toLowerCase()
    })
  }

  gemsByName = () => {
    let searchedGems = this.props.gems.filter(gem => {
      return gem.name.toLowerCase().includes(this.state.searchInput);
      })
    this.setState({
      searchedGems
    })
  }

  selectColor = (e) => {
    let clickedColor = e.target.value;
    let searchedGems = this.props.gems.filter(gem => {
      return gem.color.includes(clickedColor);
    })
    this.setState({
      searchedGems
    })
  }


  render() {
    const GemPageDisplay = this.state.GemPage ? { display: "none" } : {};
    const gemCards = this.state.searchedGems.map(gem => {
      return <GemCard 
                selectGem={this.props.selectGem} 
                key={gem.id} 
                gemId={gem.id} 
                gemName={gem.name} 
                gemFam={gem.family} 
                gemImg={gem.image} />
    })

    return (
      <div style={GemPageDisplay}>
        <div className="container">
          <Header />
          <Greeting />
          <input
            id="input-value"
            value={this.state.searchInput}
            onChange={this.updateInputValue}
            type="text"
            placeholder="Search Gem by Name"
          />
          <button onClick={this.gemsByName} 
                  className='search-button'>Find Gems!</button>
          <p className='color-dropdown'>
            <span className='search-text'>Search Gem by Color</span>
            <select onChange={this.selectColor} 
                    id='filter-colors'>
              <option value="">Select Color</option>
              <option value="White">White</option>
              <option value="Red">Red</option>
              <option value="Orange">Orange</option>
              <option value="Yellow">Yellow</option>
              <option value="Green">Green</option>
              <option value="Blue">Blue</option>
              <option value="Purple">Purple</option>
              <option value="Pink">Pink</option>
              <option value="Brown">Brown</option>
              <option value="Grey">Grey</option>
              <option value="Black">Black</option>
            </select>
          </p>
        </div>
        <div className="gem-results">{gemCards}</div>
      </div>
    );
  }
}


export default GemsPage;