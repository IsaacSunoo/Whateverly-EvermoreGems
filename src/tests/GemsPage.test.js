import React from 'react';
import GemsPage from '../components/GemsPage';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const selectGemMock = jest.fn();
const gemsByNameMock = jest.fn();
const selectColorMock = jest.fn();

const mockGem = [{
  "name": "Aquamarine",
  "id": 1,
  "color": ["Blue"],
  "family": "Beryl",
  "hardness": [7.5, 8],
  "image": "https://i.imgur.com/mupSKCl.jpg"
},
  {
    "name": "Bixbite",
    "id": 2,
    "color": ["Red"],
    "family": "Beryl",
    "hardness": [7.5, 8],
    "image": "https://i.imgur.com/mupSKCl.jpg"
  }
]

describe('GemsPage', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <GemsPage
        selectGem={selectGemMock}
        gems={mockGem}
      />
    )
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with gem color containing red', () => {
    const mockEvent = { target: { value: 'Red' } };
    wrapper.find('#filter-colors').simulate('change', mockEvent);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the proper defualt state', () => {
    expect(wrapper.state('gemPage')).toEqual(false);
    expect(wrapper.state('searchInput')).toEqual('');
  });

  it('should update the input value', () => {
    expect(wrapper.state('searchInput')).toEqual('');
    wrapper.find('#input-value').simulate('change', { target: { value: 'Aquamarine' } });
    wrapper.find('.search-button').simulate('click');
    expect(wrapper.state('searchedGems')).toEqual([mockGem[0]]);
  });

  it('should filter by color of gems', () => {
    wrapper.find('#filter-colors').simulate('change', { target: { value: 'Red' } });
    expect(wrapper.state('searchedGems')).toEqual([mockGem[1]]);
  });

});