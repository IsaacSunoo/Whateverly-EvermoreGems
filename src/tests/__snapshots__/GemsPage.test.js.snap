// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GemsPage should match the snapshot with all data passed in correctly 1`] = `
<div
  style={Object {}}
>
  <div
    className="container"
  >
    <Header />
    <Greeting />
    <input
      id="input-value"
      onChange={[Function]}
      placeholder="Search Gem by Name"
      type="text"
      value=""
    />
    <button
      className="search-button"
      onClick={[Function]}
    >
      Find Gems!
    </button>
    <p
      className="color-dropdown"
    >
      <span
        className="search-text"
      >
        Search Gem by Color
      </span>
      <select
        id="filter-colors"
        onChange={[Function]}
      >
        <option
          value=""
        >
          Select Color
        </option>
        <option
          value="White"
        >
          White
        </option>
        <option
          value="Red"
        >
          Red
        </option>
        <option
          value="Orange"
        >
          Orange
        </option>
        <option
          value="Yellow"
        >
          Yellow
        </option>
        <option
          value="Green"
        >
          Green
        </option>
        <option
          value="Blue"
        >
          Blue
        </option>
        <option
          value="Purple"
        >
          Purple
        </option>
        <option
          value="Pink"
        >
          Pink
        </option>
        <option
          value="Brown"
        >
          Brown
        </option>
        <option
          value="Grey"
        >
          Grey
        </option>
        <option
          value="Black"
        >
          Black
        </option>
      </select>
    </p>
  </div>
  <div
    className="gem-results"
  />
</div>
`;

exports[`GemsPage should match the snapshot with gem color containing red 1`] = `
<div
  style={Object {}}
>
  <div
    className="container"
  >
    <Header />
    <Greeting />
    <input
      id="input-value"
      onChange={[Function]}
      placeholder="Search Gem by Name"
      type="text"
      value=""
    />
    <button
      className="search-button"
      onClick={[Function]}
    >
      Find Gems!
    </button>
    <p
      className="color-dropdown"
    >
      <span
        className="search-text"
      >
        Search Gem by Color
      </span>
      <select
        id="filter-colors"
        onChange={[Function]}
      >
        <option
          value=""
        >
          Select Color
        </option>
        <option
          value="White"
        >
          White
        </option>
        <option
          value="Red"
        >
          Red
        </option>
        <option
          value="Orange"
        >
          Orange
        </option>
        <option
          value="Yellow"
        >
          Yellow
        </option>
        <option
          value="Green"
        >
          Green
        </option>
        <option
          value="Blue"
        >
          Blue
        </option>
        <option
          value="Purple"
        >
          Purple
        </option>
        <option
          value="Pink"
        >
          Pink
        </option>
        <option
          value="Brown"
        >
          Brown
        </option>
        <option
          value="Grey"
        >
          Grey
        </option>
        <option
          value="Black"
        >
          Black
        </option>
      </select>
    </p>
  </div>
  <div
    className="gem-results"
  >
    <GemCard
      gemFam="Beryl"
      gemId={2}
      gemImg="https://i.imgur.com/mupSKCl.jpg"
      gemName="Bixbite"
      key="2"
      selectGem={[MockFunction]}
    />
  </div>
</div>
`;
